File: utopia.py
def f(string):
    try:
        s = int(input(string))
    except:
        print("Mauvais input, veuillez réessayer")
        f(string)
    return s

k = ["Rang de l'attaqué : ", "Rang de l'attaquant : ", "Bonus/Malus : "]
l=[]
u = 1
while u == 1:
    print("Calcul de l'objectif : ")
    print("50 + (? - ?) * 5 + ?")
    for i in k:
        l.append(f(i))
    print(f"50 + ({l[0]} - {l[1]}) * 5 + {l[2]}")
    print("Objectif =", 50+(l[0] - l[1]) *5 + l[2])
    u = f("Continuer ? (Écrire 1 si oui 0 si non) : ")

File: utopia.cpp
#include <iostream>

using namespace std;

int main(){
	cout<<"Calcul de l'objectif : "<<endl;
	cout<<"50 + (? - ?)*5 + ?"<<endl;
	cout<<"Rang de l'attaqué : ";
	unsigned int rdef = 0;
	cin>>rdef;
	cout<<"50 + ("<<rdef<<" - ?)*5 + ?"<<endl;
	cout<<"Rang de l'attaquant : ";
	unsigned int ratk = 0;
	cin>>ratk;
	cout<<"50 + ("<<rdef<<" - "<<ratk<<")*5 + ?"<<endl;
	cout<<"Bonus/malus : ";
	int bonusmalus = 0;
	cin>>bonusmalus;
	cout<<"Calcul de l'objectif : "<<endl;
	cout<<"50 + ("<<rdef<<" - "<<ratk<<")*5 + "<<bonusmalus<<endl;
	cout<<"Objectif = "<<50 + (rdef - ratk) *5 + bonusmalus<<endl;
	int i = 0;
	cin>>i;
	return 0;
}
